### PR TITLE
test: improve end-to-end tests

### LINKED ISSUE #634: Improve E2E tests
- [x] Reorganize and speed up current tests
- [x] Organize test profiles on host server (valid-profile.json, invalid-json.json, invalid-geopoints.json, test-profile-{1,10}.json)
- [x]  Remove old test profiles from host server (valid-profile.json, invalid-json.json, invalid-geopoints.json, test-profile-{1,10}.json)

### PR DESCRIPTION
**Pull request by geoffturk**
Resolves #634

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,7 +53,7 @@ test:
 #--------------------------
 .PHONY: newman-test
 newman-test:
-	newman run test/e2e-tests.json -e $(ENV_FILE) --verbose --delay-request 1000
+	newman run test/e2e-tests-staging.json -e $(ENV_FILE) --verbose --delay-request 10
```
Comment (maxyzli) [inline Makefile:56]: I'm not quite sure I understand. I think the end-to-end tests are used for development, pretest, and staging. Why do we need to have a staging suffix?

**Review by maxyzli — COMMENTED** [review]
Are we planning to add tests related to feature flags later?

**Comment by geoffturk** [comment]
Yes. I want to merge this first and then start a new branch for the additional tests.

File: Makefile (hunk 1 above)
Comment (geoffturk) [reply Makefile:56]: My thinking was that we will want to add some smoke tests (e2e-tests-production) in the future so we can make sure some of the basic functionality is always working in prod.

**Review by maxyzli — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (maxyzli) [reply Makefile:56]: How about `e2e_tests_nonprod`?

**Pull request merged**